### DISCUSSION TITLE
Wonder why not use JMX collect for JVM metrics.

### DISCUSSION THREAD

**Discussion by Roiocam** [open-telemetry/opentelemetry-java-instrumentation#6548] [Q&A]
# Motivation

I am currently changing my application metrics from JMX to OTEL. Before I used `Prometheus-JMX-exporter`, they supported JVM metrics and Grafana Dashboard.

I don't see anything like this on OTEL-JAVA, but using the same way to collect JMX like `Prometheus-JMX-exporter` also works on OTEL.

I could use `Meter.batchCallback` to start JMX-Scraper work, on that small demo, everything is working fine.

JMX has a ton of metrics for JVM, I believe it could help OTEL JVM metrics complete.

**Comment by Roiocam** [comment]
Sorry for that, I notice there is already the same thing on here: https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics

**Comment by mateuszrzeszutek** [comment]
Hey @Roiocam ,

As you've already noticed, we are in fact using JMX to get the JVM metrics: https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics
We're slowly adding new JVM metrics, but since they need to be [specced](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/runtime-environment-metrics.md#jvm-metrics) it unfortunately takes time.

If you want to listen for custom JMX metrics (as defined in a config file, for example), we have an open issue for this: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/6131

**Reply by Roiocam** [reply]
Thanks for your time. 

I noticed this is not completed runtime metrics, something is missing like `memory metrics`(not memory pool), `os physical memory`, `os available processors`.

does anyone work on this? tell me if not and then I could support this.

**Reply by trask** [reply]
hi @Roiocam, we would ideally like to get new metrics added to the specification first, check out https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/runtime-environment-metrics.md#jvm-metrics

**Reply by Roiocam** [reply]
Thanks for your advice, I will enhance the specification first by opening an issue.
